pkg/pillar/zedbox/zedbox.go

```diff
@@ -33,46 +40,227 @@ import (
 	"github.com/lf-edge/eve/pkg/pillar/pubsub/socketdriver"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/sirupsen/logrus"
 )
 
-var entrypoints = map[string]func(*pubsub.PubSub){
-	"client":           client.Run,
-	"command":          command.Run,
-	"diag":             diag.Run,
-	"domainmgr":        domainmgr.Run,
-	"downloader":       downloader.Run,
-	"executor":         executor.Run,
-	"hardwaremodel":    hardwaremodel.Run,
-	"ledmanager":       ledmanager.Run,
-	"logmanager":       logmanager.Run,
-	"nim":              nim.Run,
-	"nodeagent":        nodeagent.Run,
-	"verifier":         verifier.Run,
-	"volumemgr":        volumemgr.Run,
-	"waitforaddr":      waitforaddr.Run,
-	"zedagent":         zedagent.Run,
-	"zedmanager":       zedmanager.Run,
-	"zedrouter":        zedrouter.Run,
-	"ipcmonitor":       ipcmonitor.Run,
-	"baseosmgr":        baseosmgr.Run,
-	"wstunnelclient":   wstunnelclient.Run,
-	"conntrack":        conntrack.Run,
-	"tpmmgr":           tpmmgr.Run,
-	"vaultmgr":         vaultmgr.Run,
-	"upgradeconverter": upgradeconverter.Run,
-}
+const (
+	agentName   = "zedbox"
+	socketPath  = "/var/run"
+	errorTime   = 3 * time.Minute
+	warningTime = 40 * time.Second
+)
+
+var (
+	debugOverride = false
+	//Version Set from Makefile
+	Version     = "No version specified"
+	entrypoints = map[string]func(*pubsub.PubSub){
+		"client":           client.Run,
+		"command":          command.Run,
+		"diag":             diag.Run,
+		"domainmgr":        domainmgr.Run,
+		"downloader":       downloader.Run,
+		"executor":         executor.Run,
+		"hardwaremodel":    hardwaremodel.Run,
+		"ledmanager":       ledmanager.Run,
+		"logmanager":       logmanager.Run,
+		"nim":              nim.Run,
+		"nodeagent":        nodeagent.Run,
+		"verifier":         verifier.Run,
+		"volumemgr":        volumemgr.Run,
+		"waitforaddr":      waitforaddr.Run,
+		"zedagent":         zedagent.Run,
+		"zedmanager":       zedmanager.Run,
+		"zedrouter":        zedrouter.Run,
+		"ipcmonitor":       ipcmonitor.Run,
+		"baseosmgr":        baseosmgr.Run,
+		"wstunnelclient":   wstunnelclient.Run,
+		"conntrack":        conntrack.Run,
+		"tpmmgr":           tpmmgr.Run,
+		"vaultmgr":         vaultmgr.Run,
+		"upgradeconverter": upgradeconverter.Run,
+	}
+	log *base.LogObject
+)
 
 func main() {
-	pid := os.Getpid()
-	basename := filepath.Base(os.Args[0])
-	log := base.NewSourceLogObject(basename, pid)
-	ps := pubsub.New(&socketdriver.SocketDriver{Log: log}, log)
-	if entrypoint, ok := entrypoints[basename]; ok {
-		entrypoint(ps)
+	versionPtr := flag.Bool("v", false, "Version")
+	debugPtr := flag.Bool("d", false, "Debug flag")
+	debugOverride = *debugPtr
+	if *versionPtr {
+		fmt.Printf("%s: %s\n", os.Args[0], Version)
+		return
+	}
+
+	log = agentlog.Init(agentName)
+
+	if debugOverride {
+		logrus.SetLevel(logrus.DebugLevel)
 	} else {
-		fmt.Printf("Unknown package: %s\n", basename)
+		logrus.SetLevel(logrus.InfoLevel)
+	}
+
```

Comment:
Why not put the zedbox Run code in a Run() function so you don't need to look for debug etc twice?
Basically just add "zedbox": Run to the entryPoints definition.